tests/helpers/test_template.py

```diff
@@ -475,6 +477,233 @@ def test_isnumber(hass: HomeAssistant, value, expected) -> None:
     )
 
 
+@pytest.mark.parametrize(
+    ("value", "expected"),
+    [
+        ([1, 2], True),
+        ({1, 2}, False),
+        ({"a": 1, "b": 2}, False),
+        (ReadOnlyDict({"a": 1, "b": 2}), False),
+        (MappingProxyType({"a": 1, "b": 2}), False),
+        ("abc", False),
+        (b"abc", False),
+        ((1, 2), False),
+        (datetime(2024, 1, 1, 0, 0, 0), False),
+    ],
+)
+def test_islist(hass: HomeAssistant, value, expected) -> None:
+    """Test is_list."""
+    assert (
+        template.Template("{{ is_list(value) }}", hass).async_render({"value": value})
+        == expected
+    )
+    assert (
+        template.Template("{{ value | is_list }}", hass).async_render({"value": value})
+        == expected
+    )
+    assert (
+        template.Template("{{ value is is_list }}", hass).async_render({"value": value})
```

Comment:
This is really ugly. We should name the test `list`, not `is_list`. Same for the other tests.

Reply:
I was following the `is_number` convention. Agreed it's ugly, but the advantage is that it distinguishes between the test/check and the functions to convert the value to the given type. So there are two options:

1. Leave it as is, and have two different names for converting a type vs checking a type
2. Use the same name for converting and checking. That means `list` is a test that checks the type, but is a function and filter that converts the type

Which is more desireable?

Reply:
the `number` test was already in use as a built in test, which only returns `true` if the value actually is a number (so not when it is a string
```jinja
{{ '3' is number }} # False
{{ '3' is is_number }} # True
{{ 3 is number }} # True
{{ 3 is is_number }} # True
```

Point 2 is similar with the built in function/filter `string` which converts the type, and the test `string`

Reply:
good point. Thanks!